Compile failure on Arch Linux

Arch Linux (kernel 5.12.12)
gcc (GCC) 11.1.0
clang version 12.0.0 (but it looks like it's auto-selecting GCC)

Sorry, I'm far from an expert in debugging C++ build issues, but here are the outputs, and please let me know if anything else can help.

### `cmake -B build`
```
-- The C compiler identification is GNU 11.1.0
-- The CXX compiler identification is GNU 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ju/git/rtpmidid/build
```

### `make build`
```
mkdir -p build
cd build &&	cmake .. -DCMAKE_BUILD_TYPE=Release
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ju/git/rtpmidid/build
cd build && make -j
make[1]: warning: -j0 forced in submake: resetting jobserver mode.
make[1]: Entering directory '/home/ju/git/rtpmidid/build'
make[2]: Entering directory '/home/ju/git/rtpmidid/build'
make[3]: Entering directory '/home/ju/git/rtpmidid/build'
make[3]: Entering directory '/home/ju/git/rtpmidid/build'
Consolidate compiler generated dependencies of target rtpmidid-shared
make[3]: Leaving directory '/home/ju/git/rtpmidid/build'
Consolidate compiler generated dependencies of target rtpmidid-static
make[3]: Leaving directory '/home/ju/git/rtpmidid/build'
[ 20%] Built target rtpmidid-shared
make[3]: Entering directory '/home/ju/git/rtpmidid/build'
make[3]: Entering directory '/home/ju/git/rtpmidid/build'
make[3]: Entering directory '/home/ju/git/rtpmidid/build'
[ 41%] Built target rtpmidid-static
Consolidate compiler generated dependencies of target test_poller
make[3]: Entering directory '/home/ju/git/rtpmidid/build'
make[3]: Leaving directory '/home/ju/git/rtpmidid/build'
Consolidate compiler generated dependencies of target test_rtppeer
Consolidate compiler generated dependencies of target test_rtpserver
make[3]: Leaving directory '/home/ju/git/rtpmidid/build'
make[3]: Leaving directory '/home/ju/git/rtpmidid/build'
[ 51%] Built target test_poller
Consolidate compiler generated dependencies of target rtpmidid-daemon
make[3]: Leaving directory '/home/ju/git/rtpmidid/build'
[ 62%] Built target test_rtpserver
[ 72%] Built target test_rtppeer
make[3]: Entering directory '/home/ju/git/rtpmidid/build'
[ 75%] Building CXX object src/CMakeFiles/rtpmidid-daemon.dir/main.cpp.o
[ 79%] Building CXX object src/CMakeFiles/rtpmidid-daemon.dir/rtpmidid.cpp.o
[ 82%] Building CXX object src/CMakeFiles/rtpmidid-daemon.dir/control_socket.cpp.o
In file included from /home/ju/git/rtpmidid/src/rtpmidid.cpp:23:
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:72:8: error: ‘optional’ in namespace ‘std’ does not name a template type
   72 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &hostdescription);
      |        ^~~~~~~~
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:26:1: note: ‘std::optional’ is defined in header ‘<optional>’; did you forget to ‘#include <optional>’?
   25 | #include <set>
  +++ |+#include <optional>
   26 | #include <string>
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:73:8: error: ‘optional’ in namespace ‘std’ does not name a template type
   73 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &name,
      |        ^~~~~~~~
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:73:3: note: ‘std::optional’ is defined in header ‘<optional>’; did you forget to ‘#include <optional>’?
   73 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &name,
      |   ^~~
In file included from /home/ju/git/rtpmidid/src/main.cpp:26:
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:72:8: error: ‘optional’ in namespace ‘std’ does not name a template type
   72 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &hostdescription);
      |        ^~~~~~~~
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:26:1: note: ‘std::optional’ is defined in header ‘<optional>’; did you forget to ‘#include <optional>’?
   25 | #include <set>
  +++ |+#include <optional>
   26 | #include <string>
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:73:8: error: ‘optional’ in namespace ‘std’ does not name a template type
   73 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &name,
      |        ^~~~~~~~
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:73:3: note: ‘std::optional’ is defined in header ‘<optional>’; did you forget to ‘#include <optional>’?
   73 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &name,
      |   ^~~
/home/ju/git/rtpmidid/src/rtpmidid.cpp: In constructor ‘rtpmidid::rtpmidid_t::rtpmidid_t(rtpmidid::config_t*)’:
/home/ju/git/rtpmidid/src/rtpmidid.cpp:44:16: error: ‘add_rtpmidi_client’ was not declared in this scope; did you mean ‘remove_rtpmidi_client’?
   44 |     auto res = add_rtpmidi_client(connect_to);
      |                ^~~~~~~~~~~~~~~~~~
      |                remove_rtpmidi_client
/home/ju/git/rtpmidid/src/rtpmidid.cpp:45:21: error: ‘nullopt’ is not a member of ‘std’
   45 |     if (res == std::nullopt) {
      |                     ^~~~~~~
/home/ju/git/rtpmidid/src/rtpmidid.cpp: At global scope:
/home/ju/git/rtpmidid/src/rtpmidid.cpp:51:6: error: ‘optional’ in namespace ‘std’ does not name a template type
   51 | std::optional<uint8_t>
      |      ^~~~~~~~
/home/ju/git/rtpmidid/src/rtpmidid.cpp:29:1: note: ‘std::optional’ is defined in header ‘<optional>’; did you forget to ‘#include <optional>’?
   28 | #include <rtpmidid/rtpserver.hpp>
  +++ |+#include <optional>
   29 | 
/home/ju/git/rtpmidid/src/rtpmidid.cpp: In lambda function:
/home/ju/git/rtpmidid/src/rtpmidid.cpp:213:11: error: ‘class rtpmidid::rtpmidid_t’ has no member named ‘add_rtpmidi_client’; did you mean ‘remove_rtpmidi_client’?
  213 |     this->add_rtpmidi_client(name, address, port);
      |           ^~~~~~~~~~~~~~~~~~
      |           remove_rtpmidi_client
/home/ju/git/rtpmidid/src/rtpmidid.cpp: At global scope:
/home/ju/git/rtpmidid/src/rtpmidid.cpp:222:6: error: ‘optional’ in namespace ‘std’ does not name a template type
  222 | std::optional<uint8_t>
      |      ^~~~~~~~
/home/ju/git/rtpmidid/src/rtpmidid.cpp:222:1: note: ‘std::optional’ is defined in header ‘<optional>’; did you forget to ‘#include <optional>’?
  222 | std::optional<uint8_t>
      | ^~~
make[3]: *** [src/CMakeFiles/rtpmidid-daemon.dir/build.make:118: src/CMakeFiles/rtpmidid-daemon.dir/main.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
In file included from /home/ju/git/rtpmidid/src/control_socket.cpp:28:
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:72:8: error: ‘optional’ in namespace ‘std’ does not name a template type
   72 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &hostdescription);
      |        ^~~~~~~~
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:26:1: note: ‘std::optional’ is defined in header ‘<optional>’; did you forget to ‘#include <optional>’?
   25 | #include <set>
  +++ |+#include <optional>
   26 | #include <string>
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:73:8: error: ‘optional’ in namespace ‘std’ does not name a template type
   73 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &name,
      |        ^~~~~~~~
/home/ju/git/rtpmidid/src/./rtpmidid.hpp:73:3: note: ‘std::optional’ is defined in header ‘<optional>’; did you forget to ‘#include <optional>’?
   73 |   std::optional<uint8_t> add_rtpmidi_client(const std::string &name,
      |   ^~~
/home/ju/git/rtpmidid/src/control_socket.cpp: In function ‘json rtpmidid::commands::create(rtpmidid::rtpmidid_t&, const string&, const string&, const string&)’:
/home/ju/git/rtpmidid/src/control_socket.cpp:197:29: error: ‘class rtpmidid::rtpmidid_t’ has no member named ‘add_rtpmidi_client’; did you mean ‘remove_rtpmidi_client’?
  197 |   auto alsa_port = rtpmidid.add_rtpmidi_client(name, host, port);
      |                             ^~~~~~~~~~~~~~~~~~
      |                             remove_rtpmidi_client
/home/ju/git/rtpmidid/src/control_socket.cpp:201:5: error: could not convert ‘{{"alsa_port", <expression error>}}’ from ‘<brace-enclosed initializer list>’ to ‘json’ {aka ‘nlohmann::basic_json<>’}
  201 |     };
      |     ^
      |     |
      |     <brace-enclosed initializer list>
make[3]: *** [src/CMakeFiles/rtpmidid-daemon.dir/build.make:146: src/CMakeFiles/rtpmidid-daemon.dir/rtpmidid.cpp.o] Error 1
make[3]: *** [src/CMakeFiles/rtpmidid-daemon.dir/build.make:160: src/CMakeFiles/rtpmidid-daemon.dir/control_socket.cpp.o] Error 1
make[3]: Leaving directory '/home/ju/git/rtpmidid/build'
make[2]: *** [CMakeFiles/Makefile2:192: src/CMakeFiles/rtpmidid-daemon.dir/all] Error 2
make[2]: Leaving directory '/home/ju/git/rtpmidid/build'
make[1]: *** [Makefile:101: all] Error 2
make[1]: Leaving directory '/home/ju/git/rtpmidid/build'
make: *** [Makefile:19: build/bin/rtpmidid] Error 2
```

--- a/src/rtpmidid.hpp
+++ b/src/rtpmidid.hpp
@@ -24,6 +24,7 @@
 #include <rtpmidid/poller.hpp>
 #include <set>
 #include <string>
+#include <optional>
 
 namespace rtpmidid {
 struct config_t;

